### PR TITLE
add codecov workflow

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

- GH workflow to upload test coverage reports to [app.codecov.io](https://app.codecov.io/gh/eclipse/che-dashboard)
- test coverage badge in README.md

### CODE REVIEW

File: .github/workflows/codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+name: CODECOV
+
+on:
+  push:
+    branches: 
+      - master
+  pull_request:
+    branches:
+      - master
```
Comment (sleshchenko) [inline .github/workflows/codecov.yaml:18]: Do we really need two tools for the same thing? 
It's pretty the same as https://github.com/eclipse/che-dashboard/blob/master/.github/workflows/pr.yaml#L48, no?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
-![CI](https://github.com/eclipse/che-dashboard/workflows/CI/badge.svg)
+[![CI](https://github.com/eclipse/che-dashboard/workflows/CI/badge.svg)](https://github.com/eclipse/che-dashboard/actions/workflows/ci.yaml)
+[![Codecov](https://img.shields.io/codecov/c/github/eclipse/che-dashboard)](https://app.codecov.io/gh/eclipse/che-dashboard)
```
Comment (sleshchenko) [inline README.md:2]: Don't we need to specify the specific branch?
```suggestion
[![Codecov](https://img.shields.io/codecov/c/github/eclipse/che-dashboard/master)](https://app.codecov.io/gh/eclipse/che-dashboard)
```

**Review by sleshchenko — APPROVED** [review]
+1 to add the badge as https://github.com/eclipse/che/issues/18621 suggests.

+0 to report codecov in PRs at the same time as we have https://github.com/eclipse/che-dashboard/blob/master/.github/workflows/pr.yaml#L48

File: .github/workflows/codecov.yaml (hunk 1 above)
Comment (akurinnoy) [reply .github/workflows/codecov.yaml:18]: You're right, they are pretty the same, and later I would like to get rid of `test-coverage-diff` job in favor of `codecov` as soon as we see that it provides correct reports.

File: README.md (hunk 2 above)
Comment (akurinnoy) [reply README.md:2]: I'm pretty sure that without specifying a branch it uses a default one.

**Review by olexii4 — APPROVED** [review]

**Pull request merged**